src/native/libs/System.Security.Cryptography.Native/pal_err.c

```diff
@@ -11,13 +11,15 @@ void CryptoNative_ErrClearError()
 
 uint64_t CryptoNative_ErrGetErrorAlloc(int32_t* isAllocFailure)
 {
-    unsigned long err = ERR_get_error();
+    unsigned long err = ERR_peek_last_error();
 
     if (isAllocFailure)
     {
         *isAllocFailure = ERR_GET_REASON(err) == ERR_R_MALLOC_FAILURE;
     }
 
+    // We took the one we want, clear the rest.
+    ERR_clear_error();
     return err;
 }
```

Comment:
Reviewer callout:

Rather than having the managed side keep calling CryptoNative_ErrGetErrorAlloc until it returns 0, the function now directly asks for the last error and clears the rest, reducing N+1 P/Invokes to 1.

Reply:
Renamed to CryptoNative_ErrGetExceptionError